src/connector/src/sink/kafka.rs

```diff
@@ -333,6 +333,8 @@ pub struct KafkaSinkWriter {
     schema: Schema,
     pk_indices: Vec<usize>,
     is_append_only: bool,
+    future_delivery_buffer: Vec<DeliveryFuture>,
+    max_limit: Option<usize>,
```

Comment:
No need to store as `Option<usize>`, only `usize` is fine. For unlimited, use `usize::MAX`.